File: apps/admin-v2/src/infrastructure/api/repositories/api-upload.repository.ts
import type { UploadRepository, UploadResult } from '@/domain/repositories/upload.repository';
import api from '../client';

export class ApiUploadRepository implements UploadRepository {
  async upload(file: File, folder: string): Promise<UploadResult> {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('folder', folder);

    const { data } = await api.post('/uploads', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return { url: data.url ?? data.data?.url };
  }
}

File: apps/admin_mobile/lib/infrastructure/api/repositories/api_upload_repository.dart
import 'package:dio/dio.dart';
import '../../../domain/repositories/upload_repository.dart';
import '../dio_client.dart';

class ApiUploadRepository implements UploadRepository {
  final DioClient _client;

  ApiUploadRepository(this._client);

  @override
  Future<String> upload(String filePath, String folder) async {
    final formData = FormData.fromMap({
      'file': await MultipartFile.fromFile(filePath),
      'folder': folder,
    });
    final response = await _client.dio.post('/uploads', data: formData);
    final data = response.data['data'] ?? response.data;
    return data['url'] as String;
  }
}
